### PR TITLE
CPDP-318: display api integration

### PR DESCRIPTION
**Pull request by co-category**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscribedDetails.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package uk.gov.hmrc.cgtpropertydisposalsstubs.controllers
+
+import play.api.mvc.Result
+import uk.gov.hmrc.cgtpropertydisposalsstubs.controllers.BusinessPartnerRecordController.DesBusinessPartnerRecord
+import uk.gov.hmrc.cgtpropertydisposalsstubs.controllers.SubscriptionController.SubscriptionResponse
+import uk.gov.hmrc.cgtpropertydisposalsstubs.models.{NINO, SAUTR}
+
+//case class Details(
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscribedDetails.scala]: is this file needed anymore?

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -1,22 +1,20 @@
...
 
 object AppDependencies {
 
   val compile = Seq(
-
-    "uk.gov.hmrc"             %% "bootstrap-play-26"        % "0.42.0",
-    "uk.gov.hmrc"             %% "stub-data-generator"      % "0.5.3"
+    "uk.gov.hmrc"  %% "bootstrap-play-26"        % "0.42.0",
+    "uk.gov.hmrc"  %% "stub-data-generator"      % "0.5.3",
+    "org.julienrf" %% "play-json-derived-codecs" % "3.3"
```
Comment (AndyHWChung) [inline project/AppDependencies.scala]: are you using this anywhere?

File: app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscriptionDisplayProfiles.scala [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    SubscriptionDisplay(
+      _ === "XACGTP123456789",
+      Right(individualSubscriptionDisplayDetails)
+    ),
+    SubscriptionDisplay(
+      _ === "XACGTP123456790",
+      Right(trusteeSubscriptionDisplayDetails)
+    ),
+    SubscriptionDisplay(
+      id => id.startsWith("ERG400"),
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscriptionDisplayProfiles.scala]: the form of a CGT reference number is given by the regex `^(X[A-Z]CGTP[0-9]{9})$` - could we use realistic looking CGT reference numbers in the mapping for error responses?

Comment (co-category) [reply app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscriptionDisplayProfiles.scala]: Yes we could. I was following the same design you did for the subscription profiles c.f https://github.com/hmrc/cgt-property-disposals-stubs/blob/e23b210c17acf0751d8b9e5bed094ad4a9f5e8cc/app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscriptionProfiles.scala#L120

File: project/AppDependencies.scala (hunk 2 above)
Comment (co-category) [reply project/AppDependencies.scala]: they're redundant imports now  which I used for some model case classes which have been deleted.

File: app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscribedDetails.scala (hunk 1 above)
Comment (co-category) [reply app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscribedDetails.scala]: no that needs to be removed

File: conf/prod.routes [hunk 4]
```diff
@@ -9,6 +9,7 @@ POST        /registration/02.00.00/individual                uk.gov.hmrc.cgtprop
 POST        /registration/:entityType/:idType/:idValue       uk.gov.hmrc.cgtpropertydisposalsstubs.controllers.BusinessPartnerRecordController.getBusinessPartnerRecord(entityType, idType, idValue)
 
 POST        /subscriptions/create/CGT                          uk.gov.hmrc.cgtpropertydisposalsstubs.controllers.SubscriptionController.subscribe()
+GET         /subscriptions                                     uk.gov.hmrc.cgtpropertydisposalsstubs.controllers.SubscriptionController.getSubscriptionDetails(regime : String, id : String)
```
Comment (AndyHWChung) [inline conf/prod.routes]: the URL in the DES spec doesn't have query parameters - the form of the url is `/subscriptions/{regime}/{idType}/{idValue}`

File: app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscriptionDisplayProfiles.scala (hunk 3 above)
Comment (co-category) [reply app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscriptionDisplayProfiles.scala]: changed

File: project/AppDependencies.scala (hunk 2 above)
Comment (co-category) [reply project/AppDependencies.scala]: removed redundant dep

File: app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscribedDetails.scala (hunk 1 above)
Comment (co-category) [reply app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscribedDetails.scala]: removed

File: conf/prod.routes (hunk 4 above)
Comment (co-category) [reply conf/prod.routes]: I see this in the spec: /subscriptions?regime={regimeValue}&id={idValue}

**Comment by AndyHWChung** [comment]
LGTM
